lib/imageflow/graph_runner.ex

```diff
@@ -7,8 +7,8 @@ defmodule Imageflow.GraphRunner do
          :ok <- add_outputs(job, graph.outputs),
          :ok <- send_task(job, graph),
          :ok <- save_outputs(job, graph.outputs),
-         :ok <- Native.destroy(job) do
-      :ok
+         {:ok, results} <- print_results(job, graph.outputs) do
+      if results == [], do: :ok, else: {:ok, results}
```

Comment:
I'm fine with a breaking change here, if a new result type needs to be used.

Returning different types based on the result leads to an inconsistent API.
I'd rather return `{:ok, []}`, or even `{:ok, %Result{}}` (with a result object that we could later mutate further if needed)`.

Reply:
Another option would be to just add a separate `Graph.get_results(job)` function to retrieve these values afterwards.

Might be worth checking how the .NET wrapper does it, since this one was mostly based on it

Reply:
I actually liked the `get_results` idea, but I still had to change return structure.